qlty-cli/src/commands/plugins/disable.rs

```diff
@@ -28,14 +28,19 @@ impl ConfigDocument {
     }
 
     pub fn disable_plugin(&mut self, name: &str) -> Result<()> {
-        if let Some(plugins) = self.document["plugin"].as_array_of_tables_mut() {
-            plugins.retain(|plugin| {
-                if let Some(plugin_name) = plugin.get("name").and_then(Item::as_str) {
-                    plugin_name != name
-                } else {
-                    true // Do not modify unknown data
+        let mut updated = false;
+
+        if let Some(plugin_tables) = self.document["plugin"].as_array_of_tables_mut() {
+            for plugin_table in plugin_tables.iter_mut() {
+                if plugin_table["name"].as_str() == Some(name) {
+                    updated = true;
+                    plugin_table["mode"] = value(IssueMode::Disabled.to_string());
```

Comment:
It might be a better pattern to `return Ok(())` from here and `bail()` at the end of the function as a catch-all, allowing you to drop the mutable book-keeping variable.